File: Course-Registration-System-main/java-(iteration_1)/src/CourseSection.java
import java.util.ArrayList;
// Representing specific section of a course

public class CourseSection extends Course {
    private String courseSectionCode;
    private String classroom;
    private int capacity;
    private int noOfEnrolledStudent ;
    private String day;
    private String hour;
  // Course information and section specifics
   public CourseSection(String courseName, String courseCode, int courseSemester, String courseType, int credit, ArrayList<CourseSection> courseSections, ArrayList<Course> prerequisite, Lecturer lecturer, String courseSectionCode, String classroom, int capacity,int noOfEnrolledStudent, String day, String hour) {
        super(courseName, courseCode, courseSemester, courseType, credit, courseSections, prerequisite, lecturer);
        this.courseSectionCode = courseSectionCode;
        this.classroom = classroom;
        this.capacity = capacity;
        this.noOfEnrolledStudent=noOfEnrolledStudent;
        this.day = day;
        this.hour = hour;
    }
    // Retrieve section code

      public void setCourseSectionCode(String courseSectionCode) {
        this.courseSectionCode = courseSectionCode;
    }
    // Setting section code


    //Retrieving assigned classroom

    public String getClassroom() {
        return classroom;
    }
    // Setting assigned classroom

    public void setClassroom(String classroom) {
        this.classroom = classroom;
    }
    // Retrieving section capacity

    public int getCapacity() {
        return capacity;
    }
    // Setting section capacity

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }
    // Retrieving scheduled day(s)

    public String getDay() {
        return day;
    }
    // Setting scheduled day(s)
    public void setDay(String day) {
        this.day = day;
    }
    // Retrieving scheduled hour
    public String getHour() {
        return hour;
    }
    // Setting scheduled hour

    public void setHour(String hour) {
        this.hour = hour;
    }

    public int getNoOfEnrolledStudent() {
        return noOfEnrolledStudent;
    }

    public void setNoOfEnrolledStudent(int noOfEnrolledStudent) {
        this.noOfEnrolledStudent = noOfEnrolledStudent;
    }

    public String getCourseSectionCode() {
        return courseSectionCode;
    }
    // Displaying section information, course details and section-specific details

   public String viewCourseSectionInfo(){
        return (viewCourseInfo() + "\nSection info: " +  getCourseSectionCode() + "\nClassroom: " + getClassroom() +
                "\nHour: " + getHour() + "\n");
    }
}

File: Course-Registration-System-main/python-(iteration_2)/src/CourseSection.py
from Course import Course

class CourseSection(Course):
    def __init__(self, courseName, courseCode, courseSemester, courseType, credit, courseSections, prerequisite, lecturer,
                 courseSectionCode, classroom, capacity, noOfEnrolledStudent, day, hour):
        
        super().__init__(courseName, courseCode, courseSemester, courseType, credit, courseSections, prerequisite, lecturer)
        self.__courseSectionCode = courseSectionCode
        self.__classroom = classroom
        self.__capacity = capacity
        self.__noOfEnrolledStudent = noOfEnrolledStudent
        self.__day = day
        self.__hour = hour

    def setCourseSectionCode(self, courseSectionCode):
        self.courseSectionCode = courseSectionCode

    def getClassroom(self):
        return self.__classroom

    def setClassroom(self, classroom):
        self.__classroom = classroom

    def getCapacity(self):  
        return self.__capacity

    def setCapacity(self, capacity):
        self.__capacity = capacity

    def getDay(self): 
        return self.__day

    def setDay(self, day):  
        self.__day = day

    def getHour(self): 
        return self.__hour

    def setHour(self, hour):  
        self.__hour = hour

    def getNoOfEnrolledStudents(self):  
        return self.__noOfEnrolledStudent

    def setNoOfEnrolledStudents(self, noOfEnrolledStudent):  
        self.__noOfEnrolledStudent = noOfEnrolledStudent

    def getCourseSectionCode(self):
        return self.__courseSectionCode
    
    def viewCourseSectionInfo(self):
        return (
            f"{self.viewCourseInfo()}\nSection info: {self.getCourseSectionCode()}\n"
            f"Classroom: {self.getClassroom()}\nHour: {self.getHour()}\n"
        )
        
